scripts/fix_doctests.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import re
+import sys
+from subprocess import PIPE, run
+
+import hypothesistooling as tools
+
+
+class FailingExample(object):
+
+    def __init__(self, chunk):
+        """Turn a chunk of text into an object representing the test."""
+        location, *lines = [l + '\n' for l in chunk.split('\n') if l.strip()]
+        self.location = location.strip()
+        pattern = 'File "(.+?)", line (\d+?), in .+'
+        self.file, line = re.match(pattern, self.location).groups()
+        self.line = int(line) + 1
+        got = lines.index('Got:\n')
+        self.expected_lines = lines[lines.index('Expected:\n') + 1:got]
+        self.got_lines = lines[got + 1:]
+
+    def __repr__(self):
+        return '{}\nExpected: {!r:.60}\nGot:      {!r:.60}'.format(
+            self.location, self.expected, self.got)
+
+
+def run_doctests():
```

Comment:
I think this could use some documentation about the safe/unsafe distinction. I *think* I understand why it's like that, but it's not totally clear to me why it matters if we change the line numbers and whether there's a simpler approach we could take (I assume you've thought of the obvious things and they don't work, but I'd like why spelled out).

Reply:
Sure - a change is "unsafe" if it alters the position of later changes.  The line number is already of dubious use because "Line 6 in data.rst" could actually be the sixth line of some docstring that is included in data.rst, so I didn't want to complicate things.

We *could* work around this by keeping track of how many lines were added and where, but it's simpler just to loop (I'll note that too).  We could also make an 'unsafe' change *to each file* per loop, which I will do.

Reply:
Is there a reason you can't just sort the changes in reverse order of where their first line is and apply them one at a time in that order? That way each change doesn't interfere with the starting point of the previous changes.

Reply:
Hmm, I actually could do a pre-processing pass to establish the true initial location, and then yes.  Time for some less hacky, more considered data processing 😄